src/detail/header.cpp

```diff
@@ -597,11 +597,30 @@ on_insert_transfer_encoding()
         md.transfer_encoding.codings += rv->size();
         for(auto t : *rv)
         {
-            if(! md.transfer_encoding.is_chunked)
+            auto& mte = md.transfer_encoding;
+
+            if(! mte.is_chunked )
             {
-                if(t.id == transfer_coding::chunked)
-                    md.transfer_encoding.is_chunked = true;
-                continue;
+                if( t.id == transfer_coding::chunked )
```

Comment:
"transfer coding" is not a term of art. It is Transfer-Encoding. We are inventing the word "coding" and then attaching it to the word "transfer" which is confusingly similar to the name of the field. This looks all sort of clumsy to read.

Reply:
I'm not inventing anything, this comes from the RFC:
https://datatracker.ietf.org/doc/html/rfc9112#name-transfer-codings

But I'll change it.

Reply:
> this comes from the RFC:

Hmm...yes, then it might be ok if it was only used for Transfer-Encoding. However, we know that implementations in the wild are mostly broken and use Content-Encoding instead which leaves us with a confusing name that would have to apply to both. Also note that transfer-coding includes chunked in the RFC (12.3), but our enum can't have that (nor should it). I do like a shorter name, this is looking a bit busy.